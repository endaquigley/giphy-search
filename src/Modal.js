import React, { Component } from 'react';
import styled, { keyframes } from 'styled-components';
import closeIcon from './images/close-icon.svg';

const scale = keyframes`
  from { transform: scale(0) }
  to { transform: scale(1) }
`;

const Wrapper = styled.div`
  position: fixed;
  z-index: 1;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  background: rgba(0, 0, 0, 0.90);
`;

const Close = styled.img`
  position: absolute;
  top: 20px;
  right: 20px;
  width: 25px;
  height: 25px;
`;

const Video = styled.video`
  width: 100%;
  max-width: 85vw;
  max-height: 85vh;
  animation: 0.4s ${ scale };
`;

class Modal extends Component {

  componentDidMount() {
    // disable scroll when modal is open
    document.body.classList.add('no-scroll');
  }

  componentWillUnmount() {
    // enable scroll when modal is closed
    document.body.classList.remove('no-scroll');
  }

  render() {
    return (
      <Wrapper onClick={ () => this.props.handleClick() }>
        <Close src={ closeIcon } />
        <Video
          loop
          muted
          autoPlay
          playsInline
          src={ this.props.selected.images.original.mp4 }
        />
      </Wrapper>
    )
  }

};

export default Modal;
