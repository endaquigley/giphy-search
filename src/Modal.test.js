import React from 'react';
import ReactDOM from 'react-dom';
import Modal from './Modal';

it('renders the original video when a media object is provided', () => {
  const selected = {
    id: 1,
    images: {
      preview: { mp4: 'https://enda.ie/filename-01.mp4' },
      original: { mp4: 'https://enda.ie/filename-01.mp4' }
    }
  };

  const component = document.createElement('div');
  ReactDOM.render(<Modal selected={ selected } />, component);

  const video = component.querySelector('video');
  expect(video.src).toBe('https://enda.ie/filename-01.mp4');

  expect(component).toMatchSnapshot();
});