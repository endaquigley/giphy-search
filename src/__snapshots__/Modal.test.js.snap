// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the original video when a media object is provided 1`] = `
<div>
  <div
    class="sc-bdVaJa eVvdxU"
  >
    <img
      class="sc-bwzfXH PoSZD"
      src="close-icon.svg"
    />
    <video
      class="sc-htpNat fdswIX"
      loop=""
      autoplay=""
      src="https://enda.ie/filename-01.mp4"
    />
  </div>
</div>
`;
